cpp/test/spatial/epsilon_neighborhood.cu

```diff
@@ -65,14 +72,13 @@ class EpsNeighTest : public ::testing::TestWithParam<EpsInputs<T, IdxT>> {
                                 false);
   }
 
-  void TearDown() override { RAFT_CUDA_TRY(cudaStreamDestroy(stream)); }
-
   EpsInputs<T, IdxT> param;
   cudaStream_t stream = 0;
   rmm::device_uvector<T> data;
   rmm::device_uvector<bool> adj;
   rmm::device_uvector<IdxT> labels, vd;
   IdxT batchSize;
+  const raft::handle_t handle;
```

Comment:
I noticed that here, the handle is captured by (const) value, whereas in another similar class above, the handle is captured by const reference.  If capturing by value produces correct results, would you consider just capturing the handle by (nonconst) value?  In general, capturing nonconst values instead of const values or references can help make the class `semiregular`, which makes it easier to reason about the class' behavior.  This is also consistent with `mdspan` behavior (which is pointer-like, therefore `semiregular`).

Reply:
Which similar class above are you referring to? This is a test class, which both creates and owns the handle so it's not using a reference. It's also being used in  the setup in a test function above- it could be made const there, but doesn't need to be a reference.

Reply:
Sorry, I had forgotten that this was a test (so you have to create a handle anyway) and that a default-constructed handle is meaningful : - ) .